packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -816,16 +835,26 @@ class SplineROITool extends ContourSegmentationBaseTool {
       const spline = this._updateSplineInstance(element, annotation);
       const splinePolylineCanvas = spline.getPolylinePoints();
 
-      this.updateContourPolyline(
-        annotation,
-        {
-          points: splinePolylineCanvas,
-          closed: data.contour.closed,
-          targetWindingDirection: ContourWindingDirection.Clockwise,
-        },
-        viewport,
-        { updateWindingDirection: data.contour.closed }
-      );
+      // Compare with existing contour polyline (converted to canvas)
+      const existingPolylineCanvas =
+        annotation.data.contour.polyline?.map(worldToCanvas) || [];
+
+      // Only update/invalidate if geometry actually changed
+      if (
+        !this._arePolylinesEqual(existingPolylineCanvas, splinePolylineCanvas)
+      ) {
+        const isClosed = data.contour.closed;
+        this.updateContourPolyline(
```

Comment:
Consider moving the check for is same into the updateContourPolyline itself, perhaps under a flag.
Also, the isSamePolyline with the changes I suggested works fine for either 2d or 3d point data, so you might not need to convert the old one to canvas points if the new one is being converted anyways, and might have new changes inside updateContourPolyline.